File: Programming-languages-and-laboratory/Java/Exercises/31_compito_java_2019_07_05/Cliente.java
import java.util.*;

public class Cliente {
    private String nomeCognome;
    private List<Noleggio> noleggi;

    public Cliente(String nomeCognome) {
        this.nomeCognome = nomeCognome;
        this.noleggi = new LinkedList<Noleggio>();
    }

    public void addNoleggio(String targa, int giorni, double costo) {
        noleggi.add(new Noleggio(targa, giorni, costo));
    }

    public double getMax() {
        double max = 1.0;
        for (Noleggio n : noleggi) {
            double costo = n.getTot();
            if (costo > max)
                max = costo;
        }
        return max;
    }

    public String toString() {
        return nomeCognome + " " + getMax() + noleggi.size();
    }
}

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/31_compito_java_20190705/31_GestioneVeicoli_python/Cliente.py
class Cliente:
    def __init__(self, nome_cognome):
        self._nome_cognome = nome_cognome
        self._noleggi = []
        self._costo_max = 0.0

    def add_noleggio(self, n):
        self._noleggi.append(n)

    def confronta_costo(self, c):
        if c > self._costo_max:
            self._costo_max = c

    def __str__(self):
        return self._nome_cognome + " " + str(self._costo_max) + " " + str(len(self._noleggi))
